src/main/java/filemanager/service/impl/XmlWriteServiceImpl.java

```diff
@@ -38,7 +43,11 @@ public void writeXmlFile(List<Interaction> interactions, String outputPath) thro
         XmlMapper xmlMapper = new XmlMapper();
         Feed feed = new Feed(interactions);
         xmlMapper.writeValue(file, feed);
-        counters.get("exportsAmount").inc(interactions.size());
+        Map<String, String> tags = new HashMap<>();
+        interactions.forEach(interaction -> tags.put(Tags.CLIENT, interaction.getClientName()));
```

Comment:
Aren't all your iterations already filtered to belong to a single client? If you aren't sure in this method, you may move the metric into caller method, which is actually sure to have all interactions grouped by client.